workflow/persist/sqldb/workflowRepository.go

```diff
@@ -0,0 +1,192 @@
+	FinishedAt time.Time      `db:"finishedAt"`
+}
+
+func convert(wf *wfv1.Workflow) *WorkflowDB {
+	jsonWf, _ := json.Marshal(wf)
+	startT, _ := time.Parse(time.RFC3339, wf.Status.StartedAt.Format(time.RFC3339))
+	endT, _ := time.Parse(time.RFC3339, wf.Status.FinishedAt.Format(time.RFC3339))
+	return &WorkflowDB{
+		Id:         string(wf.UID),
+		Name:       wf.Name,
+		Namespace:  wf.Namespace,
+		Workflow:   string(jsonWf),
+		Phase:      wf.Status.Phase,
+		StartedAt:  startT,
+		FinishedAt: endT,
+	}
+
+}
+
+func (wdc *WorkflowDBContext) IsInterfaceNil() bool {
+	return wdc == nil
+}
+func (wdc *WorkflowDBContext) IsSupportLargeWorkflow() bool {
+	return wdc.SupportLargeWorkflow
+}
+
+func (wdc *WorkflowDBContext) Init(sess sqlbuilder.Database) {
+	wdc.Session = sess
+}
+
+// Save will upset the workflow
+func (wdc *WorkflowDBContext) Save(wf *wfv1.Workflow) error {
+
+	if wdc != nil && wdc.Session == nil {
+		return errors.InternalError("DB is not initiallized")
+	}
+	wfdb := convert(wf)
+
+	err := wdc.update(wfdb)
+
+	if err != nil {
+		if errors.IsCode(errors.CodeDBUpdateRowNotFound, err) {
+			return wdc.insert(wfdb)
+		} else {
+			log.Warn(err)
+			return errors.InternalErrorf("Error in inserting workflow in persistence. %v", err)
+		}
+	}
+
+	log.Info("Workflow update successfully into persistence")
+
+	return nil
+}
+
+func (wdc *WorkflowDBContext) insert(wfDB *WorkflowDB) error {
+	tx, err := wdc.Session.NewTx(nil)
+	if err != nil {
+		return errors.InternalErrorf("Error in creating transaction. %v", err)
+	}
+	err = tx.Collection(wdc.TableName).InsertReturning(wfDB)
+	if err != nil {
+		return errors.InternalErrorf("Error in inserting workflow in persistence. %v", err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		return errors.InternalErrorf("Error in Committing workflow insert in persistence. %v", err)
+	}
+	return nil
+}
+
+func (wdc *WorkflowDBContext) update(wfDB *WorkflowDB) error {
+	tx, err := wdc.Session.NewTx(nil)
+	if err != nil {
+		return errors.InternalErrorf("Error in creating transaction. %v", err)
+	}
+	err = tx.Collection(wdc.TableName).UpdateReturning(wfDB)
+	if err != nil {
+		if strings.Contains(err.Error(), "upper: no more rows in this result set") {
+			return errors.DBUpdateNoRowFoundError(err, "")
+		}
+		return errors.InternalErrorf("Error in updating workflow in persistence %v", err)
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		return errors.InternalErrorf("Error in Committing workflow update in persistence %v", err)
+	}
+	return nil
+}
+
+func (wdc *WorkflowDBContext) Get(uid string) *wfv1.Workflow {
+	var wfDB WorkflowDB
+	var wf wfv1.Workflow
+	if wdc.Session == nil {
+		log.Warn("DB is not initiallized")
+		return nil
+	}
+
+	err := wdc.Session.Collection(wdc.TableName).Find("id", uid).One(&wfDB)
+	if err != nil {
+
+	}
+	if wfDB.Id != "" {
+		err := json.Unmarshal([]byte(wfDB.Workflow), &wf)
+		if err != nil {
+			log.Warn(" Workflow unmarshalling is failed for ")
```

Comment:
Incomplete log message. Please use a logrus logging contexts with the workflow name as the field label.